src/daft-dsl/src/join.rs

```diff
@@ -0,0 +1,110 @@
+use common_error::{DaftError, DaftResult};
+use daft_core::{
+    schema::{Schema, SchemaRef},
+    JoinType,
+};
+use indexmap::IndexSet;
+
+use crate::{Expr, ExprRef};
+
+/// Get the columns between the two sides of the join that should be merged in the order of the join keys.
+/// Join keys should only be merged if they are column expressions.
+pub fn get_common_join_keys<'a>(
+    left_on: &'a [ExprRef],
+    right_on: &'a [ExprRef],
+) -> impl Iterator<Item = String> + 'a {
```

Comment:
because you already have the lifetime of `'a` you can prob have this return ```impl Iterator<Item =&'a str>```

Reply:
this would allow you to avoid a clone if the downstream code doesn't need an owned string